### PR TITLE
Redux refactor

### PR DESCRIPTION
**Pull request by martinlabate**

### CODE REVIEW

File: wbooks/src/Screens/App/Screens/BookDetails/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      BooksApi.getBook(bookId).then((response) => {
+        dispatch(addBook(response.data));
+      }).catch((error) => {
+        if (error.response.status === '400') {
+          dispatch(notFound())
+        }
+      });
+    },
+    onLoading: () => {
+      dispatch(startLoading());
```
Comment (pabloferro) [inline wbooks/src/Screens/App/Screens/BookDetails/index.js]: You can dispatch this before `BooksApi.getBook(bookId).then((response) => {` and remove the `onLoading` function.

File: wbooks/src/Screens/App/Screens/BookDetails/present.js [hunk 2]
```diff
@@ -1,72 +1,42 @@
 import React, { Component } from 'react';
```
Comment (pabloferro) [inline wbooks/src/Screens/App/Screens/BookDetails/present.js]: We name the presentational components `layout.js`.
If a component doesn't have a containter, we name the presentational `index.js`

File: wbooks/src/Screens/App/Screens/BookDetails/present.js [hunk 3]
```diff
@@ -1,72 +1,42 @@
...
 import { Link } from 'react-router-dom';
-import defaultBookIcon from './default_book.svg';
-import './BookDetail.css';
-import BooksApi from './services/BooksApi'
-import BookCover from './BookCover'
+import defaultBookIcon from '../../../../Assets/default_book.svg'
```
Comment (pabloferro) [inline wbooks/src/Screens/App/Screens/BookDetails/present.js]: I think you are not using this.

File: wbooks/src/Screens/App/Screens/BookDetails/present.js [hunk 4]
```diff
@@ -1,72 +1,42 @@
...
-      } else {
-        this.setState({fetchingBook: false, error: true})
-      }
-    })
-  }
-
-  setBook = (book) => {
-    this.setState({book, fetchingBook: false})
-  }
+    this.props.onLoading();
```
Comment (pabloferro) [inline wbooks/src/Screens/App/Screens/BookDetails/present.js]: Remove this after the refactor I mentioned

File: wbooks/src/Screens/App/Screens/BookDetails/index.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+const mapStateToProps = state => {
+  return {
+    book: state.bookDetail.book,
+    notFound: state.bookDetail.notFound,
+    fetchingBook: state.bookDetail.fetchingBook
+  }
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => {
```
Comment (pabloferro) [inline wbooks/src/Screens/App/Screens/BookDetails/index.js]: If you don't use a parameter you don't need write it. (check this for all `mapDispatchToProps`)

File: wbooks/src/Screens/App/Screens/Dashboard/Screens/Books/index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    onFetchBooks: () => {
+      BooksApi.getBooks().then((response) => {
+        dispatch(addBooks(response.data));
+      });
+    },
+    onLoading: () => {
+      dispatch(startLoading());
```
Comment (pabloferro) [inline wbooks/src/Screens/App/Screens/Dashboard/Screens/Books/index.js]: Same here, you can dispatch `startLoading()` before the `getBooks()`

File: wbooks/src/Screens/App/Screens/Dashboard/index.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    onClick: () => {
+      dispatch(applyText());
+      dispatch(applyFilter());
+    },
+    onChange: (newText) => {
+      dispatch(changeText(newText));
+    },
+    onFilterChange: (filterType) => {
+      if (filterType === 'author') {
+        dispatch(filterAuthor());
```
Comment (pabloferro) [inline wbooks/src/Screens/App/Screens/Dashboard/index.js]: You could use one action with a payload instead.

File: wbooks/src/redux/Books/reducer.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import {ADD_BOOKS, START_LOADING, OPEN_BOOK} from './actions'
+
+const initialState = {
+  books: [],
+  fetchingBooks: false,
+  openBook: ""
```
Comment (pabloferro) [inline wbooks/src/redux/Books/reducer.js]: Are you using this part of the state?
If not remove all code related to `openBook`.

**Review by pabloferro — CHANGES_REQUESTED** [review]

File: wbooks/src/Screens/App/Screens/BookDetails/present.js (hunk 3 above)
Comment (martinlabate) [reply wbooks/src/Screens/App/Screens/BookDetails/present.js]: I'm using it for the fake recommendations section =)

**Review by pabloferro — APPROVED** [review]

**Pull request merged**